src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/PseudoRoleToAccessProfileGenerator.java

```diff
@@ -0,0 +1,114 @@
+        caseRoles.addAll(getCaseRolesFromAcls(stateAcls(caseTypeDefinition)));
+        caseRoles.addAll(getCaseRolesFromAcls(caseFieldAcls(caseTypeDefinition)));
+        caseRoles.addAll(getCaseRolesFromAcls(caseFieldComplexAcls(caseTypeDefinition)));
+
+        return caseRoles;
+    }
+
+    private List<String> getCaseRolesFromAcls(List<AccessControlList> accessControlLists) {
+        return accessControlLists.stream()
+            .map(AccessControlList::getAccessProfile)
+            .filter(accessProfile -> accessProfile.matches(CASE_ROLE_ID_REGEX))
+            .collect(Collectors.toList());
+    }
+
+    private List<String> getIdamRolesFromAcls(List<AccessControlList> accessControlLists) {
+        return accessControlLists.stream()
+            .map(AccessControlList::getAccessProfile)
+            .filter(accessProfile -> !accessProfile.matches(CASE_ROLE_ID_REGEX))
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> eventAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getEvents().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> stateAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getStates().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> caseFieldAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getCaseFieldDefinitions().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> caseFieldComplexAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getCaseFieldDefinitions().stream()
+            .flatMap(element -> element.getComplexACLs().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<RoleToAccessProfileDefinition> createRoleToAccessProfiles(String ctId,
+                                                                           Set<String> roles,
+                                                                           boolean addIdamPrefix) {
+        return roles.stream()
+            .map(role -> createRoleToAccessProfile(ctId, role, addIdamPrefix))
+            .collect(Collectors.toList());
+    }
+
+    private RoleToAccessProfileDefinition createRoleToAccessProfile(String ctId, String role, boolean addIdamPrefix) {
+        return RoleToAccessProfileDefinition.builder()
+            .accessProfiles(role)
+            .roleName(addIdamPrefix ? IDAM_PREFIX + role : role)
+            .caseTypeId(ctId)
+            .readOnly(false)
+            .disabled(false)
+            .authorisations(null)
```

Comment:
Guess we don't need these if these defaults are correct?
```suggestion
```

Reply:
may be we need it, as we have Boolean objects not primitives